### DISCUSSION TITLE
SSM not running by default when launching the bottlerocket instance in EC2

### DISCUSSION THREAD

**Discussion by azizzoaib786** [bottlerocket-os/bottlerocket#2448] [Q&A] [answered]
Unable to connect to EC2 using SSM, bottlerocket instance throws below error.

```         Starting Bottlerocket userdata configuration system...
[    5.571032] early-boot-config[1452]: Provider error: Unable to serialize settings from instance user data: TOML data did not contain 'settings' section```

Any idea whats missing?

**Comment by stmcginnis** [comment, marked as answer]
Hi @azizzoaib786 - I've seen the same thing if there are issues preventing the configuration files being written out. That causes some core services to fail to start and prevents being able to connect using SSM.

Are you passing in user data? From the error, it looks like the contents of user data might not be correct or complete. During boot up there is a service called `early-boot-config` that reads configuration data and uses that to format and write out various configuration files with the appropriate settings. If anything fails with that process, it is not able to complete setting up all config files and services are not able to start.

**Reply by azizzoaib786** [reply]
Yes, was able to figure out that.. Now I'm passing the user_data just for the sake purpose.

```
 [settings]
 motd = "Something!"

**Reply by wzhiana** [reply]
-@azizzoaib786 so can you share config how you start ec2 instance without being connected to ECS/EKS? 
which image do you use?-

i found my issue! all is good

**Reply by webern** [reply]
@azizzoaib786 are you saying that, at first, you had no userdata at all and that caused to the boot to fail? Then you needed to add some trivial userdata to get the boot to succeed?

If so, @stmcginnis, it confirms this issue still needs to be addressed. #1470
